### PR TITLE
chore: improve Makefile targets nim_status_client and rcc

### PR DESCRIPTION
**Pull request by michaelsbradleyjr**
Allow for `make` to check the timestamps of files relevant to those targets and thereby skip one or both targets if nothing's changed.

Related to a [discussion from yesterday](https://github.com/status-im/status-desktop/pull/3545#discussion_r709388487).

NOTE: I added some comments and commented code re: `UI_SOURCES` that I'll clean up prior to merge, after we decide what we want to do.

There are a lot of files in `ui/` such that:
```
UI_SOURCES := $(shell find ui -type f | grep -v resources\.qrc)
```
causes things to be slower (because `make` has to check so many timestamps) than simply rebuilding `resources.rcc` every time.

Using a smaller file set gives a nice speedup, but per my comment, developers will need to be aware that changing some kinds of files in `ui/` will require a manual step to get a fresh build of `resources.rcc`.

Since the `rcc` target is pretty fast these days (takes a couple+ seconds on my machine), we could leave it alone and only modify the `nim_status_client` target.

@jakubgs is it important to include directories in `NIM_SOURCES` and `UI_SOURCES` so that removal of a file can be detected by `make`? That probably wouldn't make much difference for `NIM_SOURCES` but maybe it's more of a concern for `UI_SOURCES`.

Note that with the improved `nim_status_client` target in place, it's practical to have the `run` target depend on that one instead of `rcc`, which is nice because you no longer need to do `make` and then `make run`, you can just do `make run` and everything needed to run the app will be re/built (or not) appropriately.

### CODE REVIEW

**Review by pinebit — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -219,13 +219,31 @@ fleets-remove:
...
+# UI_SOURCES := $(shell find ui -type f | grep -v resources\.qrc)
+
+# This list is much smaller and the rcc target completes very quickly if
+# there's nothing to be done. However, if there's a change to e.g. a PNG file
+# then it will be necessary to delete resources.rcc (or touch a .qml file,
+# etc.) in order to trigger a fresh build of resources.rcc
+
+UI_SOURCES := $(shell find ui -type f \
+		| grep -E "qmldir|\.(qml|qrc)\$$" \
+		| grep -v resources\.qrc)
```
Comment (jakubgs) [inline Makefile:236]: You can do the same thing with just `find`, without having to invoke two more processes:
```sh
 > find ui -type f -regextype egrep -iregex '.*(qmldir|qml|qrc)$' -not -iname 'resources.qrc' | wc -l
484

 > find ui -type f | grep -E "qmldir|\.(qml|qrc)\$$" | grep -v resources.qrc | wc -l         
484
```

**Review by jakubgs — APPROVED** [review]
Nice to see proper use of `make`.

File: Makefile (hunk 1 above)
Comment (michaelsbradleyjr) [reply Makefile]: Thanks, change made, though I did need to account for BSD `find` not having the `-regextype` option (it has `-E` instead).

Comment (jakubgs) [reply Makefile]: Oh, that's annoying.

**Pull request merged**